File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.py
import sys
input = sys.stdin.readline

case = int(input())

position = []

for _ in range(case):
    pos = list(map(int, input().split()))
    position.append(pos)

position.sort(key=lambda x: (x[1], x[0]))

for i in position:
    print(i[0], i[1])

File: 백준/Silver/11651. 좌표 정렬하기 2/좌표 정렬하기 2.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [, ...targetArr] = input;
const result = targetArr.map((data) => data.split(" ").map(Number));

result.sort((a, b) => {
  if (a[1] === b[1]) {
    return a[0] - b[0]
  }
  return a[1] - b[1]
})

console.log(result.map(data => data.join(" ")).join("\n"))
